test/integration/reexecuting-domain-handlers.test.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * This test covers a bug where the state gets out of sync when an
+ * action is pushed within another action. We first encountered this
+ * on a project where we wanted to remove a "focused" state when saving
+ * data using another action.
+ */
+
+import Microcosm from '../../src/microcosm'
+
+describe('Re-executing domain handlers', function() {
+  it('does not call the same domain handler twice if it would result in no change', function() {
+    let repo = new Microcosm()
+
+    let addOne = jest.fn((count, n) => count + n)
+    let addTwo = jest.fn((count, n) => count + n)
+
+    repo.addDomain('counterOne', {
+      getInitialState() {
+        return 0
+      },
+      register() {
+        return { addOne }
+      }
+    })
+
+    repo.addDomain('counterTwo', {
+      getInitialState() {
+        return 0
+      },
+      register() {
+        return { addTwo }
+      }
+    })
+
+    let one = repo.append('addOne')
+
+    repo.push('addTwo', 2)
+
+    one.resolve(1)
+
+    expect(addOne).toHaveBeenCalledTimes(1)
+
+    expect(addTwo).toHaveBeenCalledTimes(1)
+
+    expect(repo.state).toEqual({ counterOne: 1, counterTwo: 2 })
+  })
+
+  it('does not call the same domain handler twice when two domains listen to the same action from a different key', function() {
```

Comment:
this test in particular i'm having a tough time following, the "same action from a different key" doesn't line up with what i see below, also why each action is called 3 times and not less is a mystery to me